substrate/runtime-support/src/storage/mod.rs

```diff
@@ -39,6 +39,22 @@ impl<'a> Input for IncrementalInput<'a> {
 	}
 }
 
+// TODO: only introduce this wrapper for types where it makes sense, ideally have it within the module declaration.
+
+struct AppendZeroes<'a> {
+	input: &'a mut Input,
+}
+
+impl<'a> Input for AppendZeroes<'a> {
+	fn read(&mut self, into: &mut [u8]) -> usize {
+		let r = self.input.read(into);
+		for z in &mut into[r..] {
```

Comment:
will panic if the slice was completely fillet, won't it?

Reply:
should just return an empty slice

Reply:
interesting artifact of the slicing syntax... that `&[][0]` would panic but `&[][0..]` wouldn't.